From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pierre=20Sandstro=CC=88m?= <pierre.sandstrom@dstny.com>
Date: Thu, 15 Dec 2022 23:41:32 +0100
Subject: Add payload type mappings

This patch ensures that we assign the following payload types for our external codecs.
115: AMR-WB
114: AMR
18: G729

diff --git a/media/engine/payload_type_mapper.cc b/media/engine/payload_type_mapper.cc
index c63d1d72212b086aedab07169602982d66102cb0..7cae6a8bdaad0fb863235acc7be450d368eb52c8 100644
--- a/media/engine/payload_type_mapper.cc
+++ b/media/engine/payload_type_mapper.cc
@@ -57,7 +57,9 @@ PayloadTypeMapper::PayloadTypeMapper()
            {{"G728", 8000, 1}, 15},
            {{"DVI4", 11025, 1}, 16},
            {{"DVI4", 22050, 1}, 17},
-           {{"G729", 8000, 1}, 18},
+           {{"G729", 8000, 1, {{"annexb", "no"}}}, 18},
+           {{"AMR", 8000, 1, {{"mode-change-capability", "2"}, {"max-red", "0"}}}, 114},
+           {{"AMR-WB", 16000, 1, {{"mode-change-capability", "2"}, {"max-red", "0"}}}, 115},
 
            // Payload type assignments currently used by WebRTC.
            // Includes data to reduce collisions (and thus reassignments)
